src/templates/blog-post.tsx

```diff
@@ -46,7 +46,10 @@ export const Head = ({
   const { author, seoMetaText, title, publicationDate, heroImage } =
     data.contentfulBlogPost;
 
-  const shareImagePath = heroImage.shareImage.resize.src;
+  let shareImagePath = '';
+  if (heroImage && heroImage.shareImage && heroImage.shareImage.resize) {
+    shareImagePath = heroImage.shareImage.resize.src;
+  }
```

Comment:
please replace with a more idiomatic fix: 

```suggestion
  const  shareImagePath = heroImage.shareImage?.resize.src;
```